Directory.Build.targets

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+
+  <Target
+    Name="EnsureJavaScriptCodeUpToDate"
+    BeforeTargets="BeforeBuild"
+    Inputs="$(MSBuildThisFileDirectory)yarn.lock"
+    Outputs="$(RootIntDir)\JsUpToDateCheck\yarn.lock"
```

Comment:
>Js [](start = 27, length = 2)

Should it be Yarn or node modules? I.e. not JavaScript, but something node_module oriented. Also, it is really about checking the native build. Maybe MSBuildNodeModuleVersionCheck? Too long?

Reply:
msbuild is ONLY timestamp based, if we go into node_modules msbuild would be spending way to much time. ALso we would have to 'copy' all the node_modules files... -or- we'd have to create a custom tool that scans all files computes hashes and then compare against previous run. 

I'd say too much work and not worth it then... 
This should be as fast as possible and catch 90% of use cases (whcih is after syncing) If we are going after correctness we need a differetn build engine because msbuild has massive holes in correctness :) You tell me how many times you have to use 'rebuild' or 'clean' to get into a good state :P

Reply:
And given all the symlinks in node_modules pointing back into the vnext folder where we have all our *.*proj files and code this would never ever be incremetnal and run on almost every build....

Reply:
Per above, I've added package.json files...